lib/foreman_inventory_upload/async/delayed_start.rb

```diff
@@ -0,0 +1,46 @@
+module ForemanInventoryUpload
+  module Async
+    module DelayedStart
+      extend ActiveSupport::Concern
+
+      START_WINDOW = 3.hours.seconds
+
+      Poll = Algebrick.atom
+
+      def after_delay(delay = nil, logger: nil, &block)
+        logger ||= self.logger if respond_to? :logger
+        delay ||= ForemanRhCloud.requests_delay || Random.new.rand(START_WINDOW)
+        delay = delay.to_i
+
+        logger&.debug("planning a delay for #{delay} seconds before the rest of the execution")
+
+        sequence do
+          plan_self(delay: delay)
+          concurrence(&block)
+        end
+      end
+
+      def run(event = nil)
+        case event
+        when nil
+          plan_event(Poll, delay) if delay > 0
```

Comment:
Technically yes, but the idea is that the task itself doesn't do any "useful" work, so it will finish the execution immediately. If the events are passed quickly enough, I won't oppose to removing the `if` part and sending the event immediately.

Reply:
If I'm reading it right it will get suspended on the next line and never wake up. Either the event should be always sent or the action should only suspend itself if the event is being sent